### ISSUE TITLE
[Bug]: Getting Started Fails

### ISSUE THREAD

**Issue by usmcamp0811** [tweag/jupyenv#533]
### Current Behavior

I used the template and did `nix run` and get the following error. 

```
╰─🚧 nix run .\#        
[2024-02-29 21:13:42 jupyenv] needs to build JupyterLab.
Traceback (most recent call last):
  File "/nix/store/kzndcs6zwrsih1mnyf8x11hgh5yfy9c5-python3.10-jupyterlab-3.6.1/bin/.jupyter-lab-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_server/extension/application.py", line 598, in launch_instance
    subapp = _preparse_for_subcommand(cls, args)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_server/extension/application.py", line 38, in _preparse_for_subcommand
    app.initialize_subcommand(subc, subargv)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/traitlets/config/application.py", line 113, in inner
    return method(app, *args, **kwargs)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/traitlets/config/application.py", line 726, in initialize_subcommand
    self.subapp.initialize(argv)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/traitlets/config/application.py", line 113, in inner
    return method(app, *args, **kwargs)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_core/application.py", line 252, in initialize
    self.migrate_config()
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_core/application.py", line 181, in migrate_config
    migrate()
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_core/migrate.py", line 241, in migrate
    with open(os.path.join(env["jupyter_config"], "migrated"), "w", encoding="utf-8") as f:
OSError: [Errno 30] Read-only file system: '/nix/store/ka67xr3wflf6r3x26mplrzc665y1lix4-jupyter-dir/config/migrated'
```

### Expected Behavior

Jupyter Lab starts?

### Steps To Reproduce

```
nix flake init --template github:tweag/jupyenv
nix run .#
```

### OS

```shell
NixOS
```

### Version

23.11

**Comment by usmcamp0811** [comment]
So I've been playing with this some this evening and I've been able to get it to work on 3 other computers, 2 running Nix and the third running Arch. I have a feeling this is a shell environment variable problem but I'm not certain and its odd cause all my Nix boxes are based on the same flake. If I get it on the problem system working I'll report what I did.

**Comment by tomasel** [comment]
I've encountered the same problem and haven't been able to find a solution yet.
The only thing I can think of is that I executed `nixos-rebuild --update switch` right before `nix run` and the output I got is the exact same as for @usmcamp0811 :

```
[2024-11-04 10:53:05 jupyenv] needs to build JupyterLab.
Traceback (most recent call last):
  File "/nix/store/kzndcs6zwrsih1mnyf8x11hgh5yfy9c5-python3.10-jupyterlab-3.6.1/bin/.jupyter-lab-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_server/extension/application.py", line 598, in launch_instance
    subapp = _preparse_for_subcommand(cls, args)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_server/extension/application.py", line 38, in _preparse_for_subcommand
    app.initialize_subcommand(subc, subargv)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/traitlets/config/application.py", line 113, in inner
    return method(app, *args, **kwargs)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/traitlets/config/application.py", line 726, in initialize_subcommand
    self.subapp.initialize(argv)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/traitlets/config/application.py", line 113, in inner
    return method(app, *args, **kwargs)
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_core/application.py", line 252, in initialize
    self.migrate_config()
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_core/application.py", line 181, in migrate_config
    migrate()
  File "/nix/store/i2ihlpzkbcysdyq8srhh3xlpmv6cskgy-python3-3.10.9-env/lib/python3.10/site-packages/jupyter_core/migrate.py", line 241, in migrate
    with open(os.path.join(env["jupyter_config"], "migrated"), "w", encoding="utf-8") as f:
OSError: [Errno 30] Read-only file system: '/nix/store/ka67xr3wflf6r3x26mplrzc665y1lix4-jupyter-dir/config/migrated'
```

**Comment by GTrunSec** [comment]
closes by #524
